iiblib/iib_build_details_model.py

```diff
@@ -1,135 +1,278 @@
 class IIBBuildDetailsModel(object):
     """Model class handling data about index build task"""
 
-    def __init__(
-        self,
-        _id,
-        state,
-        reason,
-        state_history,
-        from_index,
-        from_index_resolved,
-        bundles,
-        removed_operators,
-        organization,
-        binary_image,
-        binary_image_resolved,
-        index_image,
-        request_type,
-        arches,
-        bundle_mapping,
-        omps_operator_version,
-    ):
-        """
-        Args:
-            _id (int)
-                Id of build
-            state (str)
-                State of build
-            state (str)
-                Reason for state change
-            from_index (str)
-                Reference of index image used as source for rebuild
-            from_index_resolved (str)
-                Reference of new index image
-            bundles (list)
-                List of bundles to be added to index image
-            removed_operators (list)
-                List of operators to be removed from index image
-            organization (str)
-                Name of organization to push to in the legacy app registry
-            binary_image (str)
-                Reference of binary image used for rebuilding
-            binary_image_resolved (str)
-                Checksum reference of binary image that was used for rebuilding
-            index_image (str)
-                Reference of index image to rebuild
-            request_type (str)
-                Type of iib build task (add or remove)
-            arches (list)
-                List of architectures supported in new index image
-            bundle_mapping (dict)
-                Operator names in "bundles" map to: list of "bundles" which
-                map to the operator key
-            omps_operator_version (dict)
-                Operator version returned from OMPS API call used for Add request
-        """
-        self.id = _id
-        self.state = state
-        self.reason = reason
-        self.state_history = state_history
-        self.from_index = from_index
-        self.from_index_resolved = from_index_resolved
-        self.bundles = bundles
-        self.removed_operators = removed_operators
-        self.organization = organization
-        self.binary_image = binary_image
-        self.binary_image_resolved = binary_image_resolved
-        self.index_image = index_image
-        self.request_type = request_type
-        self.arches = arches
-        self.bundle_mapping = bundle_mapping
-        self.omps_operator_version = omps_operator_version
+    __slots__ = [
```

Comment:
The \_\_slots\_\_ are defined here just because it is the same for every model but you will never generate base IIBBuildDetailsModel  it will always end up at specific model where are the attributes defined. 
A thought is to add the attribute description here and remove it from specifics models or another thought is to add the description here and it will be duplicated in IIBBuildDetailsModel and specifics models. What do you think @lipoja

Reply:
For me those parameters are missing here for future users of this class we should let them know that these parameters are accepted by the class and its subclasses.

More transparent would be to have it on both places. In those subclasses I would mention what is defined in IIBBuildDetailsModel in the description. But on the other hand it creates duplicity and every change should be updated on multiple places, which is not good practice.

In this case I would vote for easier maintenance and document only relevant parameters for each class. Mentioning that some parameters are inherited from parent class.

Reply:
Description added to the base model and all the specific models